src/app/admin/onprem-administration/yourlicense/yourlicense.controller.js

```diff
@@ -38,26 +40,82 @@ export class OnPremisesAdminYourLicenseCtrl {
           }
         }
       }
+    }, (error)=> {
+      this.isLoading = false;
+      this.cheNotification.showError(error.data.message ? error.data.message : 'Installation manager server error.');
     });
 
-    this.updateLicenseState();
+    this.checkLicense();
+    this.errorMessage = null;
   }
 
+
   /**
-   * Update license's state
+   * Update license state
    */
   updateLicenseState() {
+    this.islicenseExpired = !this.license.properties || this.license.properties.isExpired === 'true';
+    this.licenseState = 'LICENSE';
+    this.newLicense = angular.copy(this.license.key);
+    this.maxUsers = this.imsLicenseApi.getNumberOfAllowedUsers();
+  }
+
+
+  /**
+   * Check license
+   */
+  checkLicense() {
     if (this.license.key) {
-      this.licenseState = 'LICENSE';
-      this.maxUsers = this.imsLicenseApi.getNumberOfAllowedUsers();
+      this.updateLicenseState();
     } else {
       this.imsLicenseApi.fetchLicense().then(() => {
-        this.licenseState = 'LICENSE';
-        this.maxUsers = this.imsLicenseApi.getNumberOfAllowedUsers();
+        this.updateLicenseState();
       }, () => { //if no license
         this.licenseState = 'NO_LICENSE';
+        this.newLicense = null;
       });
     }
   }
 
+
+  /**
+   * Delete current  license
+   */
+  deleteLicense() {
+    let promise = this.imsLicenseApi.deleteLicense();
+
+    this.isLoading = true;
+    promise.then(()=> {
+      this.isLoading = false;
+      this.cheNotification.showInfo('License successfully deleted.');
+      this.licenseState = 'NO_LICENSE';
+      this.newLicense = null;
+    }, (error)=> {
+      this.isLoading = false;
+      this.cheNotification.showError(error.data.message ? error.data.message : 'License server error.');
+    });
+  }
+
+
+  /**
+   * Add new license
+   */
+  addLicense() {
+    let promise = this.imsLicenseApi.addLicense(this.newLicense);
+
+    this.errorMessage = null;
+    this.isLoading = true;
+    promise.then(()=> {
+      this.isLoading = false;
+      this.cheNotification.showInfo('License successfully added.');
+      this.imsLicenseApi.fetchLicenseProperties().then(()=> {
+        this.checkLicense();
+      });
+    }, (error)=> {
+      this.isLoading = false;
+      this.errorMessage = error.data.message ? error.data.message : 'License server error.';
```

Comment:
should we reset this.errorMessage = null in other methods like deleteLicense() ?